facebook/OpenBIC
# fby3.5: bb: Support sensor reading

fby3.5: bb: Support sensor reading
Summary:
- Modify BB BIC sensor number.
- Modify BB BIC sensor table.
- Modify BB BIC sensor threshold.
- Fix ADM1278 sensor reading lose float value problem.

Dependency: #246 

Test plan:
- Build code: Pass
- Get sensor reading: Pass

Log:
1. Get BB sensor reading.
root@bmc-oob:~# sensor-util slot1 --thre | grep BB
BB_INLET_TEMP                (0xD1) :   26.43 C     | (ok) | UCR: 50.00 | UNC: NA | UNR: 150.00 | LCR: NA | LNC: NA | LNR: NA
BB_OUTLET_TEMP               (0xD2) :   29.43 C     | (ok) | UCR: 55.00 | UNC: NA | UNR: 150.00 | LCR: NA | LNC: NA | LNR: NA
BB_HSC_TEMP                  (0xD3) :   25.95 C     | (ok) | UCR: 55.00 | UNC: NA | UNR: 125.00 | LCR: NA | LNC: NA | LNR: NA
BB_P5V_STBY                  (0xD4) :    5.14 Volts | (ok) | UCR: 5.55 | UNC: 5.50 | UNR: 5.65 | LCR: 4.45 | LNC: 4.50 | LNR: 4.15
BB_P12V_STBY                 (0xD5) :   12.36 Volts | (ok) | UCR: 13.29 | UNC: 13.17 | UNR: 14.30 | LCR: 10.71 | LNC: 10.77 | LNR: 10.08
BB_P3V3_STBY                 (0xD6) :    3.36 Volts | (ok) | UCR: 3.57 | UNC: 3.54 | UNR: 3.73 | LCR: 3.04 | LNC: 3.07 | LNR: 2.74
BB_P5V_USB                   (0xE9) :    5.14 Volts | (ok) | UCR: 5.40 | UNC: 5.35 | UNR: 5.65 | LCR: 4.60 | LNC: 4.65 | LNR: 4.15
BB_P1V2_STBY                 (0xD9) :    1.21 Volts | (ok) | UCR: 1.30 | UNC: 1.28 | UNR: 1.36 | LCR: 1.10 | LNC: 1.12 | LNR: 1.00
BB_P1V0_STBY                 (0xDA) :    1.01 Volts | (ok) | UCR: 1.08 | UNC: 1.07 | UNR: 1.13 | LCR: 0.92 | LNC: 0.93 | LNR: 0.83
BB_MEDUSA_VIN                (0xDB) :   12.22 Volts | (ok) | UCR: 13.88 | UNC: 13.75 | UNR: 14.00 | LCR: 11.12 | LNC: 11.25 | LNR: 9.25
BB_MEDUSA_VOUT               (0xD7) :   12.22 Volts | (ok) | UCR: 13.88 | UNC: 13.75 | UNR: 14.00 | LCR: 11.12 | LNC: 11.25 | LNR: 9.25
BB_HSC_VIN                   (0xDC) :   12.19 Volts | (ok) | UCR: 13.29 | UNC: 13.17 | UNR: 14.30 | LCR: 10.71 | LNC: 10.77 | LNR: 10.08
BB_MEDUSA_CUR                (0xDF) :   66.15 Amps  | (ok) | UCR: 144.00 | UNC: NA | UNR: NA | LCR: NA | LNC: NA | LNR: NA
BB_HSC_IOUT                  (0xDE) :    6.03 Amps  | (ok) | UCR: 22.96 | UNC: NA | UNR: 31.92 | LCR: NA | LNC: NA | LNR: NA
BB_FAN_IOUT                  (0xE8) :    0.40 Amps  | (ok) | UCR: 14.56 | UNC: NA | UNR: 39.26 | LCR: NA | LNC: NA | LNR: NA
BB_MEDUSA_PWR                (0xD8) :  807.64 Watts | (ok) | UCR: 1800.00 | UNC: NA | UNR: NA | LCR: NA | LNC: NA | LNR: NA
BB_HSC_PIN                   (0xDD) :   73.23 Watts | (ok) | UCR: 287.10 | UNC: NA | UNR: 398.46 | LCR: NA | LNC: NA | LNR: NA
BB_HSC_EIN                   (0xEA) :   73.60 Watts | (ok) | UCR: NA | UNC: NA | UNR: NA | LCR: NA | LNC: NA | LNR: NA
BB_HSC_PEAK_IOUT             (0xEB) :    7.18 Amps  | (ok) | UCR: NA | UNC: NA | UNR: NA | LCR: NA | LNC: NA | LNR: NA
BB_HSC_PEAK_PIN              (0xEC) :   89.43 Watts | (ok) | UCR: NA | UNC: NA | UNR: NA | LCR: NA | LNC: NA | LNR: NA

2. Check ADM1278 sensor value is correct.
- Before fix
root@bmc-oob:~# sensor-util slot1 | grep HSC
HSC Temp                     (0xE) :   41.00 C     | (ok)
HSC Input Vol                (0x29) :   12.00 Volts | (ok)
HSC Output Cur               (0x30) :    9.00 Amps  | (ok)
HSC Input Pwr                (0x39) :  109.00 Watts | (ok)
BB_HSC_TEMP                  (0xD3) :   26.00 C     | (ok)
BB_HSC_VIN                   (0xDC) :   12.00 Volts | (ok)
BB_HSC_IOUT                  (0xDE) :    6.00 Amps  | (ok)
BB_HSC_PIN                   (0xDD) :   73.00 Watts | (ok)
BB_HSC_EIN                   (0xEA) :   73.00 Watts | (ok)
BB_HSC_PEAK_IOUT             (0xEB) :    7.00 Amps  | (ok)
BB_HSC_PEAK_PIN              (0xEC) :   89.00 Watts | (ok)

- After fix
root@bmc-oob:~# sensor-util slot1 | grep HSC
HSC Temp                     (0xE) :   40.71 C     | (ok)
HSC Input Vol                (0x29) :   12.17 Volts | (ok)
HSC Output Cur               (0x30) :    9.12 Amps  | (ok)
HSC Input Pwr                (0x39) :  109.23 Watts | (ok)
BB_HSC_TEMP                  (0xD3) :   26.19 C     | (ok)
BB_HSC_VIN                   (0xDC) :   12.18 Volts | (ok)
BB_HSC_IOUT                  (0xDE) :    6.12 Amps  | (ok)
BB_HSC_PIN                   (0xDD) :   74.15 Watts | (ok)
BB_HSC_EIN                   (0xEA) :   73.50 Watts | (ok)
BB_HSC_PEAK_IOUT             (0xEB) :    7.18 Amps  | (ok)
BB_HSC_PEAK_PIN              (0xEC) :   89.43 Watts | (ok)
